### PR TITLE
feat(feedback): Add github feedback components

### LINKED ISSUE #57778: GitHub Feedback Component
Let's do a PoC of a feedback component that enables you to give feedback in a public way (github discussion).

We likely want two components:
- Button
  - button with GitHub icon
  - accepts discussion URL as a prop
  - is `ExternalLink`
  - accepts buttonProps so that consumers can modify it  
- Tooltip
  - you can wrap it around anything
  - accepts discussion URL as a prop
  - accepts tooltipProps so that consumers can modify it
  - users can disable it if they are using it in a render map
  - the link lives inside the tooltip text, so the tooltip needs to be `hoverable`
 
By default, both of them should be visible only to early adopters (unless you specify otherwise via props).

### PR DESCRIPTION
**Pull request by ArthurKnaus**
Add pre-styled components to  ask users for feedback on GitHub.
Add `GithubFeedbackButton`
Add `GithubFeedbackTooltip`

Screenshot 2023-10-10 at 13 12 28
Screenshot 2023-10-10 at 13 12 36

- Closes https://github.com/getsentry/sentry/issues/57778

### CODE REVIEW

**Review by matejminar — APPROVED** [review]

File: static/app/components/githubFeedbackButton.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import {t} from 'sentry/locale';
+
+type GithubFeedbackButtonProps = LinkButtonProps & {href: string};
+
+const title = t('Give us feedback on GitHub');
+
+export function GithubFeedbackButton(props: GithubFeedbackButtonProps) {
+  return (
+    <Tooltip title={title}>
+      <LinkButton size="sm" external icon={<IconGithub />} {...props} />
```
Comment (matejminar) [inline static/app/components/githubFeedbackButton.tsx:13]: Should the text of the button be optional and default to `t('Give Feedback')`?

Comment (ArthurKnaus) [reply static/app/components/githubFeedbackButton.tsx:13]: hmm, then icon-only usage is not possible anymore but maybe we never want this 🤔

Comment (matejminar) [reply static/app/components/githubFeedbackButton.tsx:13]: That's a good point. Would you be able to pass null for children if you want just an icon button?

Comment (ArthurKnaus) [reply static/app/components/githubFeedbackButton.tsx:13]: You could but then the buttons typescript types will not force you to add an `aria-label`.

**Pull request merged**